docs/book/src/tasks/experimental-features/machine-pools.md

```diff
@@ -25,3 +25,17 @@ More details on `MachinePool` can be found at:
 [MachinePool CAEP](https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/proposals/20190919-machinepool-api.md)
 
 For developer docs on the MachinePool controller, see [here](./../../developer/architecture/controllers/machine-pool.md).
+
+## MachinePools vs MachineDeployments
+
+Although MachinePools provide a similar feature to MachineDeployments, MachinePools do so by leveraging an InfraMachinePool which corresponds 1:1 with a resource like VMSS on Azure or Autoscaling Groups on AWS which we treat as a black box. When a MachinePool is scaled up, the InfraMachinePool scales itself up and populates its provider ID list based on the response from the infrastructure provider. On the other hand, a when a MachineDeployment is scaled up, new Machines are created which then create an individual InfraMachine, which corresponds to a VM in Azure.
```

Comment:
> which corresponds to a VM in Azure

this makes it sound like Machine is an Azure only thing, might be better to stay generic on the infra, or to say "for example"